### PR TITLE
Fixes #7478

### LINKED ISSUE #7478: [Integration]: Integration with OrientDB
### Is there an existing integration?

- [X] I have searched the existing integrations.

### Use Case

The goal of this issue is to build an integration with OrientDB. [OrientDB](https://www.orientdb.org/) is an open-source NoSQL multi-model database which helps organizations to unlock the power of graph databases without deploying multiple systems to handle other data types. This helps you to increase performance and security while supporting scalability.

### Resources 💡

🏁 To get started, please refer to the following resources:

[Rest API docs for OrientDB](https://orientdb.org/docs/3.1.x/misc/OrientDB-REST.html) 
[Building a Database Handler Docs](https://docs.mindsdb.com/contribute/data-handlers): Documentation provides a guide on how to build an integration for a database.
Orient DB docs: https://www.orientdb.org/docs/3.0.x/
JDBC drivers for integration: https://www.orientdb.org/integration

### Next Steps 🕵️‍♂️ 🕵️‍♀️

- You can familiarize yourself with the OrientDB API.
- Create a new handler that connects MindsDB to a OrientDB.
- Implement the core features mentioned above.
- Thoroughly test the integration to ensure proper functionality.
- Submit a pull request with the new OrientDB Integration handler.

### 📢 Additional rewards 🏆

Each App integration brings  :two: :five:  : points for SWAG and entry into the draw for a 👇 :

💻 Razer Blade 16 Laptop
👕 🐻 MindsDB Swag
ℹ️ For more info check out https://mindsdb.com/hacktoberfest/ 👈

### PR DESCRIPTION
**Pull request by victorian177**
## Description
Summary of changes made:
- Created a handler for **OrientDB**.
- Implemented functions for connecting to the **OrientDB** database i.e. `connect`, `disconnect`, and `check_connection`.
- Implemented functions for querying the **OrientDB** database i.e. `query`, `native_query`, `get_tables`, and `get_columns`.
- Implemented a unit test class for the **OrientDB** handler.

Fixes #7478

## Type of change
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [x] 📄 This change requires a documentation update

## Verification Process
To ensure the changes are working as expected:
 - [x]   Test Location: mindsdb/mindsdb/integrations/handlers/orientdb_handler/tests/test_orientdb_handler.py
 - [x]   Verification Steps: Run the following command in MindsDB Web Editor
 ```SQL
 CREATE DATABASE example_display_name  -- display name for database.
WITH ENGINE = 'orientdb',             -- name of the MindsDB handler
PARAMETERS = {
  "user": "root",          -- Your database user.
  "password": "root",      -- Your password.
  "host": "127.0.0.1",     -- Host (It can be an IP or a URL).
  "port": 2424,            -- Default port is 2424.
  "database": "example_db" -- The name of your database.
};
```
It requires that you have an instance of **OrientDB** server running accessed through `localhost:2480` to create the database and tables.
Run: `docker run -d -p 2424:2424 -p 2480:2480 orientdb:2.2`.

## Checklist:
- [x] My code follows the style guidelines(PEP 8) of MindsDB by running the `black` Python formatter.
- [x] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

## Some things to note
- The Python wrapper for OrientDB i.e. `pyorient` only works with **OrientDB** version 2.2. `pyorient` was last updated in 2017.

### CODE REVIEW

**Comment by victorian177** [comment]
I keep getting this [error message](https://github.com/mindsdb/mindsdb/actions/runs/6439663582/job/17487606776?pr=7616). It states that it is a problem with the tests folder.

File: docker/executor.Dockerfile [hunk 1]
```diff
@@ -15,7 +15,7 @@ RUN pip install ".[grpc]" ".[telemetry]"
 
 RUN pip install git+https://github.com/mindsdb/lightwood.git@staging --upgrade --no-cache-dir
 # Install our app
-# COPY ./mindsdb /mindsdb/mindsdb
+# COPY ../mindsdb /mindsdb/mindsdb
```
Comment (ZoranPandovski) [inline docker/executor.Dockerfile:18]: Please exclude this File from the PR

File: mindsdb/integrations/handlers/orientdb_handler/orientdb_handler.py [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+    def connect(self) -> StatusResponse:
+        """
+        Connect to the OrientDB database.
+
+        Returns:
+            StatusResponse: A response object indicating the success of the connection.
+        """
+
+        status_response = StatusResponse()
+        if self.is_connected is False:
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/orientdb_handler/orientdb_handler.py:53]: Can we avoid nested try/except please? Use separate functions to handle the client creation and db_open

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Thanks. Please check the comments and also:
* Remove Manual QA file
*  Exclude the changes in HTTP API and the unit/ml handlers tests

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Please make sure your branch is up to date with staging and exclude all files where you didn't add changes

**Comment by ZoranPandovski** [comment]
Please let's re-open another PR, it is hard to review this one with 8000 lines changed

**Pull request closed without merging**